src/compiler/checker.ts

```diff
@@ -11378,6 +11380,33 @@ module ts {
             return undefined;
         }
 
+        function getSymbolFromDeclarationKeyword(node: Node): Symbol {
+            switch(node.kind) {
+                case SyntaxKind.ClassKeyword:

```

Comment:
Ok, shall I only special case it on anonymous functions too? Because if we enable it on `function name() {}` then to be consistent — all declaration keywords should have quick info.